src/Features/Core/Portable/ExtractMethod/AbstractSyntaxTriviaService.Result.cs

```diff
@@ -110,15 +110,9 @@ private int GetFirstEndOfLineIndex(List<SyntaxTrivia> list)
                                             location => location,
                                             location => resolver(root, location, _annotations[location]));
 
-                // check variable assumption. ordering of two pairs can't be changed
                 Contract.ThrowIfFalse(
-                    tokens[TriviaLocation.BeforeBeginningOfSpan].RawKind == 0 /* && don't care */ ||
-                    /* don't care && */ tokens[TriviaLocation.AfterEndOfSpan].RawKind == 0 ||
-                    tokens[TriviaLocation.BeforeBeginningOfSpan].Span.End <= tokens[TriviaLocation.AfterEndOfSpan].SpanStart);
```

Comment:
this condition was actually bogus.  Consider something simple like `list?(0)` where the user selects `(0)`, in this case, the before will be `?` and the after will be the start of the following statement.  However, because we're going to extract all of `list?(0)` out, it will be the case that that can move after the start of that that following statement.

Reply:
i want to actually rewrite this entire type.  but i viewed it as out of scope for this PR>